dashboard-all-the-things/src/app/dashboard-carousel/dashboard-carousel.component.ts

```diff
@@ -1,51 +1,64 @@
+import { ReviewComponent } from '../dashboards/review';
+import { SonarCoverageComponent } from '../dashboards/sonar-coverage';
+import { StackOverflowComponent } from '../dashboards/stackoverflow/stackoverflow.component';
+import { WidgetEvent } from '../dashboards/WidgetEvent';
+import { DynamicComponent } from './dynamic.component';
+import { SocketService } from './socket.service';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ConnectableObservable, Subscription } from 'rxjs';
+
+let COMPONENTS = [
+  { key: "coverage", type: SonarCoverageComponent },
+  { key: "ciWall", type: ExternalUrlComponent },
+  { key: "reviews", type: ReviewComponent },
+  { key: "stackoverflow", type: StackOverflowComponent }
+];
 
 @Component({
   selector: 'app-dashboard-carousel',
   templateUrl: 'dashboard-carousel.component.html',
-  styleUrls: ['dashboard-carousel.component.scss']
+  styleUrls: ['dashboard-carousel.component.scss'],
+  entryComponents: COMPONENTS.map((el) => el.type)
+    .filter((elem, index, arr) => arr.indexOf(elem) === index)
 })
 export class DashboardCarouselComponent implements OnInit, OnDestroy {
 
-  private CAROUSEL_CURRENT_POSITION = '--carousel-current-position';
-  static ROTATE_RATE_IN_MILLISECONDS = 3 * 1000;
+  socketService: SocketService;
+  connection: Subscription;
+
+  type: any;
+  event: WidgetEvent;
 
-  animation: Subscription;
-  position: number;
-  dasboardCount: number = 3;
-  element: ElementRef;
+  @ViewChild(DynamicComponent) dynamicComponent: DynamicComponent;
 
-  constructor(element: ElementRef) {
-    this.element = element;
-    this.position = 1;
+  constructor(socketService: SocketService) {
+    this.socketService = socketService;
   }
 
   ngOnInit(): void {
-    this.animation = Observable
-      .timer(DashboardCarouselComponent.ROTATE_RATE_IN_MILLISECONDS, DashboardCarouselComponent.ROTATE_RATE_IN_MILLISECONDS)
-      .subscribe(() => {
-        this.showNextDashboard();
+    let observable: ConnectableObservable<any> = this.socketService.create('http://localhost:3002');
+    this.connection = observable
+      .subscribe((event: WidgetEvent) => {
+        if (event.widgetKey === undefined) {
+          return;
+        }
+        let res = this.findComponentFor(event.widgetKey);
+        if (res != undefined) {
+          this.type = res.type;
+          this.event = event;
+        }
       });
+    observable.connect();
   }
 
-  private showNextDashboard() {
-    if (this.dasboardCount > this.position) {
-      this.position += 1;
-    } else {
-      this.position = 1;
-    }
-    this.setContributorPosition(this.position);
-  }
-
-  private setContributorPosition(position: number) {
-    this.element.nativeElement.style.setProperty(this.CAROUSEL_CURRENT_POSITION, position);
+  findComponentFor(key: string) {
+    return COMPONENTS.filter((elem, index, arr) => elem.key === key)[0];
```

Comment:
1) It's not Node, 😆 
2) Object.values is a valid JS function, hack it big time with `Object['values'](COMPONENTS)` because it's Friday and we can't be bothered